### DISCUSSION TITLE
v7 - reuseMaps

### DISCUSSION THREAD

**Discussion by mhendlich** [uber/react-map-gl#1722] [General]
Thanks for all the improvements which came with the v7 rewrite!
I noticed that the `reuseMaps` prop was removed from this version. Although it was marked as experimental, it served my project very well.
Are there any plans to reimplement that feature?

**Comment by Pessimistress** [comment]
Do you see a significant performance difference with and without `reuseMaps`?

The old feature was partially designed to address an old mapbox-gl issue where it did not release all resources upon `remove()`. The prop had its limitations (e.g. you can't change any initialization options) and required some non-trivial hack to simulate the load/style events when remounted.

**Reply by mhendlich** [reply]
I mostly enjoyed the fact that it helped me keep costs down in a specific scenario where a typical user flow would lead to very frequent map component mounting/unmounting, which cannot be solved in an easy way. Without the "cache", each initialization would be billed.

**Reply by Pessimistress** [reply]
Thank you for the feedback. `reuseMaps` is added back in v7.0.4

**Reply by mhendlich** [reply]
Awesome! Thanks for your great work :)

**Comment by fknop** [comment]
Hello!
I don't know if it's the best place to ask this:

I'm having an issue with the `reuseMaps` prop when I have two maps in 2 different pages (different size, etc.), when navigating from one page to the other, the second map keeps the size of the first map and doesn't fill its container.


Unfortunately I couldn't manage to reproduce it for every map that is displayed our web app.
I tried a manual `resize` call and using an AutoSizer, but it does nothing. The only thing that resizes the map properly seems to be a resize event from the browser.

Would there be a way to be able to pass a string to the `reuseMaps` prop as a sort of `id` where a map could only reuse a map instance for that id?


We rely a lot on Mapbox/maps in our apps so saving billable events is important for us.

**Reply by Pessimistress** [reply]
Can you create a CodeSandbox that reproduces this issue?

**Reply by fknop** [reply]
I'll try to reproduce it in a codesandbox and send you a link asap! :)

**Reply by fknop** [reply]
Hi, I have not had much time to look at it but I can't seem to be able to reproduce it in a codesandbox yet. I'll dig deeper in our app code and come back to you if I manage to reproduce it :)

**Reply by fknop** [reply]
@Pessimistress Hello again!
I still haven't manage to reproduce it in codesandbox, but I recorded the behavior I'm talking about. I've blurred some parts of the app to only display the maps.


First here is the video with the `reuseMaps` prop. When navigating to another page with another map and going back, the map doesn't resize properly.
I've added a red background to the parent container to show this. The canvas container takes the proper width but only the canvas is wrongly sized.

https://user-images.githubusercontent.com/6775689/177003160-fd2a1567-99b0-4d84-ad00-0e5e82eb2f40.mov

And here is a video without `reuseMaps`, everything works properly. 

https://user-images.githubusercontent.com/6775689/177003155-03f2ba1c-dc3e-4cb8-b869-0dacf7f756d2.mov

**Reply by fknop** [reply]
I managed to pinpoint the issue. It seems that explicitly passing undefined as latitude/longitude causes an error which stops the map from resizing.
